src/common/BaseManager.ts

```diff
@@ -11,9 +12,59 @@ export type Props = {
     store: Store;
 };
 
-export default interface BaseManager {
+export interface Manager {
     destroy(): void;
     exists(parentEl: HTMLElement): boolean;
     render(props: Props): void;
     style(styles: Partial<CSSStyleDeclaration>): CSSStyleDeclaration;
 }
+
+export default class BaseManager implements Manager {
+    location: number;
+
+    reactEl: HTMLElement;
+
+    constructor({ location = 1, referenceEl }: Options) {
+        this.location = location;
+        this.reactEl = this.insert(referenceEl);
+
+        this.decorate();
+    }
+
+    decorate(): void {
```

Comment:
Should this be added to the interface?

Reply:
The caveat is that `BaseManager` is the base for all managers that mount a React UI, the exception being `HighlightCreatorManager` which implements the `Manager` interface but doesn't need any UI. So that's my reasoning for leaving `decorate` out, just as we leave out `insert` from the `Manager` interface